lib/realtime/server.ex

```diff
@@ -30,64 +25,37 @@ defmodule Realtime.Server do
   @spec default_name() :: GenServer.name()
   def default_name(), do: Realtime.Server
 
-  @spec start_link(opts()) :: GenServer.on_start()
-  def start_link(opts) do
-    {start_link_opts, server_opts} = Keyword.split(opts, [:name])
-    GenServer.start_link(__MODULE__, server_opts, start_link_opts)
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(start_link_opts) do
+    GenServer.start_link(__MODULE__, [], start_link_opts)
   end
 
   @doc """
   The subscribing process will get a message when there's new data, with the form
-  {:new_realtime_data, vehicles()}
+  {:new_realtime_data, vehicles_by_route_id()}
   """
-  @spec subscribe(Route.id(), GenServer.server()) :: vehicles()
+  @spec subscribe(Route.id(), GenServer.server()) :: vehicles_by_route_id()
   def subscribe(route_id, server \\ nil) do
     server = server || default_name()
     {registry_key, vehicles} = GenServer.call(server, {:subscribe, route_id})
     Registry.register(Realtime.Registry, registry_key, route_id)
     vehicles
   end
 
+  @spec update_vehicles(vehicles_by_route_id()) :: term()
+  def update_vehicles(vehicles, server \\ __MODULE__) do
+    GenServer.cast(server, {:update_vehicles, vehicles})
```

Comment:
Now that Concentrate is taking care of the data, `Realtime.Server` is now just a pubsub server, except it doesn't even handle its own pubsubing, because it uses Registry. Would it be possible to cut out the middle man entirely? 

Some ideas for how to do that:
* Subscribers join the GenStage system somehow
* Concentrate handles the Registry
* The `realtime/server.ex` module continues to exist in order to wrap the calls to `Registry`, but without being a GenServer (and probably getting renamed).

Somebody somewhere will need to remember the state in order to give it to new subscribers as a response to their join request. Right now, that's all this genserver state does. Is Concentrate keeping track of that state anyway right now?

Reply:
That is interesting. I'm mulling it over…

My first instinct is that a pubsub server with state kind of seems like an appropriate amount of responsibility for one module.

Reply:
Concentrate is not currently retaining any state. Both for division of responsibility and also considering the future "concentrate becomes a shared library" world, I'm disinclined to change that.

Reply:
The merge table maintains the current state, although not necessarily in the format you want.

Reply:
Ah right, I forgot about that, thanks.

Reply:
`Concentrate.Consumer.VehiclePositions` is a server that's not using its state, so insourcing the `Realtime.Server` state into there might work.

Reply:
`VehiclePositions` is not currently a server. It simply consumes the merged data, translates it to the format the client expects, and passes it to the realtime server.